chalice/build_deploy_config.sh

```diff
@@ -60,11 +60,14 @@ for var in $(echo $env_json | jq -r keys[]); do
     cat "$config_json" | jq .stages.$stage.environment_variables.$var="$val" | sponge "$config_json"
 done
 
-if [[ ${CI:-} == true ]]; then
-    account_id=$(aws sts get-caller-identity | jq -r .Account)
-    export iam_role_arn="arn:aws:iam::${account_id}:role/dss-${stage}"
-    cat "$config_json" | jq .manage_iam_role=false | jq .iam_role_arn=env.iam_role_arn | sponge "$config_json"
-fi
-
+account_id=$(aws sts get-caller-identity | jq -r .Account)
+export iam_role_arn="arn:aws:iam::${account_id}:role/dss-${stage}"
+cat "$config_json" | jq .manage_iam_role=false | jq .iam_role_arn=env.iam_role_arn | sponge "$config_json"
 cat "$iam_policy_template" | envsubst '$DSS_S3_BUCKET $DSS_S3_CHECKOUT_BUCKET $dss_es_domain $account_id $stage' > "$policy_json"
 cp "$policy_json" "$stage_policy_json"
+
+if [[ ${CI:-} != true ]]; then
+	# IAM policies must be updated from an operators machine, this will not run on CI environments.
+	echo "updating $iam_role_arn with $stage_policy_json"
+	aws iam put-role-policy --role-name $lambda_name --policy-name $lambda_name --policy-document file://$stage_policy_json
```

Comment:
What's the advantage of this over `cat "$config_json" | jq .manage_iam_role=true`?

Reply:
For a given stage `dev`;
we dont use `auto-gen-policies` in our deployments script, by setting `manage_iam_role:true` we would end up with a policy that is called `dss-dev-function_handler`; there is no problem with this, the only issue is that when the CI uploads its manually managed IAM policy it uses `policy-dev.json`, but calls uses the arn name `dss-{dev}`. 

By always having a manually managed IAM Role we can keep using the same name `dss-dev` but only update the policy when we are not in a CI system. 

:)